packages/web/src/routes/session.tsx

```diff
@@ -148,6 +160,9 @@ function Session(props: {
   });
 
   async function onDeleteCell(cell: CellType | GenerateAICellType) {
+    if (!channel) {
```

Comment:
Wondering if it'd be cleaner to extract all these useEffects into a component that wraps `Session`. Then, we only render the wrapper when readOnly is false?

Reply:
I will think about this and see if I can come up with a way. Your suggestion makes sense but I'm not sure that it on its own would be sufficient to catch all the cases.

Reply:
I think I've got something here, but I'm going to make a follow on pull request for it.